storage/libradb/src/lib.rs

```diff
@@ -620,17 +623,16 @@ impl LibraDB {
         };
         let li_version = ledger_info_with_sigs.ledger_info().version();
         assert!(latest_tree_state.version >= li_version);
-        let current_epoch = if ledger_info_with_sigs
+        let mut target_epoch = ledger_info_with_sigs.ledger_info().epoch();
+        if ledger_info_with_sigs
             .ledger_info()
             .next_validator_set()
             .is_some()
         {
-            ledger_info_with_sigs.ledger_info().epoch() + 1
-        } else {
-            ledger_info_with_sigs.ledger_info().epoch()
-        };
+            target_epoch += 1;
+        }
```

Comment:
All these seem way too complex. Do I remember right that the only purpose is to get the current validator set? Can you at least add a bit comment? (I can clean up later.)